eZ/Publish/Core/Repository/Tests/LocationResolver/PermissionAwareLocationResolverTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Repository\Tests\LocationResolver;
+
+use eZ\Publish\API\Repository\Exceptions\UnauthorizedException;
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\Core\Repository\LocationResolver\PermissionAwareLocationResolver;
+use eZ\Publish\Core\Repository\Values\Content\Location;
+use PHPUnit\Framework\TestCase;
+
+final class PermissionAwareLocationResolverTest extends TestCase
```

Comment:
A test with there is no published version yet is missing. Also, `NotFoundException` could be checked

Reply:
> A test with there is no published version yet is missing. Also, `NotFoundException` could be checked

Indeed, but you're touching a deeper issue - `null` mainLocation should be handled specifically, because it's expected to be `int` when passed to `loadLocation` - as mentioned in the diff comment above.

Reply:
Added missing tests: https://github.com/ezsystems/ezpublish-kernel/pull/3050/commits/a9643dd492feebf5078463656ee0463b092051ab